### PR TITLE
Evp cleanup

### PR DESCRIPTION
**Pull request by vbatts**

### CODE REVIEW

File: evp/evp_cipher.go [hunk 1]
```diff
@@ -3,49 +3,59 @@ package evp
...
+func newCipher(cipher *C.EVP_CIPHER) *Cipher {
+	if cipher == nil {
 		return nil
 	}
-	return &Cipher{self}
+	return &Cipher{cipher}
 }
-func (self *Cipher) Nid() int {
-	return int(C.EVP_CIPHER_nid(self.evp_cipher))
+func (cipher *Cipher) Nid() int {
```
Comment (shanemhansen) [inline evp/evp_cipher.go]: oh damn, you caught be pythoning. This must be old code.

**Comment by shanemhansen** [comment]
Sorry you have to deal with my crazy old project, presumably in production. If there are any enhancements you're interested in, let me know.

File: evp/evp_cipher.go (hunk 1 above)
Comment (vbatts) [inline evp/evp_cipher.go]: heh

**Comment by vbatts** [comment]
@shanemhansen one enhancement for sure. While I would love to have a drop-in for `crypto/x509` and `crypto/tls`, those APIs are hard to line up. But in the interim, I had started working on making an `http.RoundTripper` that could be used to override the golang `http.DefaultTransport`

my last efforts were here -> https://github.com/vbatts/gossl/tree/vbatts-x509

**Pull request merged**